addons/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/discovery/internal/ModbusDiscoveryService.java

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.discovery.internal;
+
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.openhab.binding.modbus.discovery.ModbusDiscoveryParticipant;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Discovery service for Modbus bridges.
+ *
+ * This service acts as a rendezvous point between the different Modbus endpoints and any
+ * bundles that implement auto discovery through an endpoint.
+ *
+ * New bridges (TCP or Serial Modbus endpoint) should register with this service. This is
+ * handled automatically by the ModbusEndpointDiscoveryService.
+ * Also any bundles that perform auto discovery should register a ModbusDiscoveryParticipant.
+ * This ModbusDiscoveryParticipants will be called by the service when
+ * a discovery scan is requested.
+ *
+ * @author Nagy Attila Gabor - initial contribution
+ *
+ */
+@Component(immediate = true, service = DiscoveryService.class, configurationPid = "discovery.modbus")
+public class ModbusDiscoveryService extends AbstractDiscoveryService {
+
+    private final Logger logger = LoggerFactory.getLogger(ModbusDiscoveryService.class);
+
+    // Set of services that support Modbus discovery
+    private final Set<ModbusThingHandlerDiscoveryService> services = new CopyOnWriteArraySet<>();
+
+    // Set of the registered participants
+    private final Set<ModbusDiscoveryParticipant> participants = new CopyOnWriteArraySet<>();
+
+    // Set of the supported thing types. This is a union of all the thing types
+    // supported by the registered discovery services.
+    private final Set<ThingTypeUID> supportedThingTypes = new CopyOnWriteArraySet<>();
+
+    private static final int SEARCH_TIME = 5;
```

Comment:
I would prefer to have the time unit written out, e.g. `SEARCH_TIME_SECS`

Reply:
good point, renamed it.